packages/apps/plugins/plugin-markdown/src/MarkdownPlugin.tsx

```diff
@@ -244,135 +221,49 @@ export const MarkdownPlugin = (): PluginDefinition<MarkdownPluginProvides> => {
       },
       surface: {
         component: ({ data, role, ...props }, forwardedRef) => {
-          // TODO(wittjosiah): Ideally this should only be called when the editor is actually being used.
-          //   We probably want a better pattern for splitting this surface resolver up.
           const doc =
             data.active instanceof DocumentType
               ? data.active
               : data.object instanceof DocumentType
                 ? data.object
                 : undefined;
-          const space = doc && getSpace(doc);
-          const extensions = useMemo(() => {
-            // TODO(wittjosiah): Autocomplete is not working and this query is causing performance issues.
-            // const query = space?.db.query(Filter.schema(DocumentType));
-            // query?.subscribe();
-            return getCustomExtensions(doc /*, query */);
-          }, [doc, space, settings.values.editorInputMode, getViewMode(doc?.id)]);
-
-          const dispatch = useIntentDispatcher();
-          const handleCommentSelect = useCallback(() => {
-            void dispatch({ action: LayoutAction.SET_LAYOUT, data: { element: 'complementary', state: true } });
-          }, [dispatch]);
-
-          const fileManagerPlugin = useResolvePlugin(parseFileManagerPlugin);
-          const handleFileUpload = useMemo(() => {
-            if (space === undefined) {
-              return undefined;
-            }
-
-            if (fileManagerPlugin?.provides.file.upload === undefined) {
-              return undefined;
-            }
-
-            return async (file: File) => {
-              return fileManagerPlugin?.provides?.file?.upload?.(file, space);
-            };
-          }, [fileManagerPlugin, space]);
-
-          const handleViewModeChange = useCallback(
-            (viewMode: EditorViewMode) => {
-              if (doc) {
-                state.values.viewMode[doc.id] = viewMode;
-              }
-            },
-            [doc?.id],
-          );
 
           switch (role) {
-            // TODO(burdon): Normalize layout (reduce variants).
+            case 'section':
             case 'article': {
-              if (doc) {
+              if (doc && doc.content) {
                 return (
                   <DocumentEditor
-                    viewMode={getViewMode(doc.id)}
-                    toolbar={settings.values.toolbar}
                     document={doc}
-                    extensions={extensions}
-                    onCommentSelect={handleCommentSelect}
-                    onFileUpload={handleFileUpload}
-                    onViewModeChange={handleViewModeChange}
+                    extensionProviders={state.values.extensionProviders}
+                    settings={settings.values}
+                    viewMode={getViewMode(doc.id)}
+                    onViewModeChange={(viewMode: EditorViewMode) => {
```

Comment:
since doc is defined outside the switch I think I'd leave this out there too so it can be wrapped in `useCallback` the downstream effect of recreating this every render worry me

Reply:
nit: put callback as last prop (also I'd put role/coordinate at the top)

Reply:
I disagree with the callbacks-last style in this case, rather here I prefer callbacks are paired with the prop they control, which is true here. Callbacks-last might be reasonable for a single-element component like Button which don’t have controllable state, but that isn’t always the case and thus not always helpful IME.

We could put controlled props last and satisfy both styles *sometimes*, but my preference would interleave callbacks with their props.

Reply:
This is further complicated by another preference (of mine) in prop order, that flags occur last.